Samples/StorageProviders/StorageProviders/BaseJSONStorageProvider.cs

```diff
@@ -15,6 +17,13 @@ namespace Samples.StorageProviders
     /// </summary>
     public abstract class BaseJSONStorageProvider : IStorageProvider
```

Comment:
I think in order to include `BaseJSONStorageProvider` we need at least to real providers extending it. Then we could easily move it to OrleansProviders.

A provider for Mongo would have to be a separate assembly and a NuGet package because of its dependency on Mongo libraries. I am hesitant for to take a responsibility for maintaining a Mongo provider without seeing significant usage of it inside or in the community. We could move it to OrleansContrib, which is our staging ground for such things.